utils/src/main/java/software/amazon/awssdk/utils/AttributeMap.java

```diff
@@ -83,10 +83,14 @@ public AttributeMap copy() {
     }
 
     @Override
-    public void close() {
-        attributes.values().forEach(v -> IoUtils.closeIfCloseable(v, null));
-        attributes.values().forEach(this::shutdownIfExecutorService);
-    }
+    public void close() { 
+     attributes.values().stream()
```

Comment:
I would comment why we need to explicitely exclude ExecutorService

```suggestion
        // Starting from JDK 19, ExecutorService implements AutoCloseable and will await termination on close
        // When ExecutorService is provided by the user, it can result in thread deadlocks if the ExecutorService
        // is not closed from another thread. ExecutorService are shutdown without waiting for termination
     attributes.values().stream()
```